troposphere/static/js/modals/project/addImage.js

```diff
@@ -15,6 +16,10 @@ define(function (require) {
       ModalHelpers.renderModal(ProjectAddImageModal, props, function (project, image) {
         // Call the ProjectAction directly after confirmation.
         actions.ProjectActions.addResourceToProject(image, project, {silent: false});
+        // Delete the cached query - Its results are invalid now.
+        delete stores.ProjectImageStore.queryModels["?application__id="+image.id];
```

Comment:
This line captures something Steve noted to me. The current Flux-sen implement does not incorporate or provide an approach for deleting or removing cached results from queries. As we consider changes to include a "networked resource(s)" approach - we might want to include, mull, dully consider, contemplate the lifecycle of cached networked resources (/cc @cdosborn)